great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -102,14 +103,12 @@ def _query_for_year_and_month(
     # after construction. We may be able to use a hook in a property setter.
     from great_expectations.execution_engine import SqlAlchemyExecutionEngine
 
-    assert isinstance(
-        table_asset.datasource.execution_engine, SqlAlchemyExecutionEngine
-    )
+    assert isinstance(sql_asset.datasource.execution_engine, SqlAlchemyExecutionEngine)
 
-    with table_asset.datasource.execution_engine.engine.connect() as conn:
+    with sql_asset.datasource.execution_engine.engine.connect() as conn:
         datetimes: DatetimeRange = query_datetime_range(
             conn,
-            table_asset.table_name,
+            sql_asset.data_from(),
```

Comment:
@billdirks While not part of this change, the calculated variable names `year` and `month` are misleading.  They should be `years` and `months` based on the fact that they represent ranges.  Thank you.